api/src/com/cloud/storage/Volume.java

```diff
@@ -52,7 +52,8 @@
         UploadInProgress("Volume upload is in progress"),
         UploadError("Volume upload encountered some error"),
         UploadAbandoned("Volume upload is abandoned since the upload was never initiated within a specificed time"),
-        Attaching("The volume is attaching to a VM");
+        AttachingFromReady("The volume is attaching to a VM from Ready State"),
+        AttachingFromAllocated("The volume is attaching to a VM from Allocated State");
```

Comment:
I think we can still fix the issue without introducing a new state. All we're saying is to allow volumes's attaching state whether their previous state is Ready or Allocated.

Reply:
The reason we need a new state is that if an attach fails we want the disk to return to the original state it was in - Ready -> Attaching -> FAIL -> Ready, or Allocated -> Attaching -> FAIL -> Allocated.

Perhaps I am misunderstanding something?

Reply:
Looking back at the original patch, it seems that simple Allocated->Attaching would suffice. I'll change.